packages/block-editor/src/store/actions.js

```diff
@@ -658,6 +658,302 @@ export const synchronizeTemplate = () => ( { select, dispatch } ) => {
 	dispatch.resetBlocks( updatedBlockList );
 };
 
+function mapRichTextSettings( attributeDefinition ) {
+	const {
+		multiline: multilineTag,
+		__unstableMultilineWrapperTags: multilineWrapperTags,
+		__unstablePreserveWhiteSpace: preserveWhiteSpace,
+	} = attributeDefinition;
+	return {
+		multilineTag,
+		multilineWrapperTags,
+		preserveWhiteSpace,
+	};
+}
+
+/**
+ * Delete the current selection.
+ *
+ * @param {boolean} isForward
+ */
+export const deleteSelection = ( isForward ) => ( {
+	registry,
+	select,
+	dispatch,
+} ) => {
+	const selectionAnchor = select.getSelectionStart();
+	const selectionFocus = select.getSelectionEnd();
+
+	if ( selectionAnchor.clientId === selectionFocus.clientId ) return;
+
+	// Remove the blocks if the whole blocks are selected.
+	if ( ! selectionAnchor.attributeKey || ! selectionFocus.attributeKey ) {
+		dispatch.removeBlocks( select.getSelectedBlockClientIds() );
+		return;
+	}
+
+	if (
+		typeof selectionAnchor.offset === 'undefined' ||
+		typeof selectionFocus.offset === 'undefined'
+	)
+		return;
+
+	const anchorRootClientId = select.getBlockRootClientId(
+		selectionAnchor.clientId
+	);
+	const focusRootClientId = select.getBlockRootClientId(
+		selectionFocus.clientId
+	);
+
+	// It's not mergeable if the selection doesn't start and end in the same
+	// block list. Maybe in the future it should be allowed.
```

Comment:
is this specific to block list?
if that's so, couldn't two blocks from different block lists have the same `rootClientId === ''`?

it's also about nested block then isn't it? if so, would this be clearer to say something like "…start and end in the same parent block"?

Reply:
No two block lists can have the same root client ID I think.

Reply:
okay, it seemed like if they were created independently then each would have the empty string as the root id. thanks for double-checking.